TB crash on deleted row from postgres

A user reported this, and provided the traceback at the end of this description.

If I'm reading this correctly and getting the order of events correct, it seems like the connection to postgres was disconnected, which may have caused a nullpointererror deeper inside the Debezium code.

---

```text
 org.apache.kafka.connect.errors.ConnectException: org.postgresql.util.PSQLException: Database connection failed when writing to copy
 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.commitOffset(PostgresStreamingChangeEventSource.java:283)
 	at io.debezium.pipeline.ChangeEventSourceCoordinator.commitOffset(ChangeEventSourceCoordinator.java:127)
 	at io.debezium.connector.common.BaseSourceTask.commit(BaseSourceTask.java:230)
 	at io.debezium.embedded.EmbeddedEngine.commitOffsets(EmbeddedEngine.java:945)
 	at io.debezium.embedded.EmbeddedEngine.run(EmbeddedEngine.java:839)
 	at io.debezium.embedded.ConvertingEngineBuilder$2.run(ConvertingEngineBuilder.java:170)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:834)
 Caused by: org.postgresql.util.PSQLException: Database connection failed when writing to copy
 	at org.postgresql.core.v3.QueryExecutorImpl.flushCopy(QueryExecutorImpl.java:1081)
 	at org.postgresql.core.v3.CopyDualImpl.flushCopy(CopyDualImpl.java:28)
 	at org.postgresql.core.v3.replication.V3PGReplicationStream.updateStatusInternal(V3PGReplicationStream.java:193)
 	at org.postgresql.core.v3.replication.V3PGReplicationStream.forceUpdateStatus(V3PGReplicationStream.java:111)
 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.doFlushLsn(PostgresReplicationConnection.java:496)
 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.flushLsn(PostgresReplicationConnection.java:489)
 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.commitOffset(PostgresStreamingChangeEventSource.java:276)
 	... 11 more
 Caused by: java.io.IOException: Stream closed
 	at java.base/sun.nio.cs.StreamEncoder.ensureOpen(StreamEncoder.java:45)
 	at java.base/sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:152)
 	at java.base/java.io.OutputStreamWriter.flush(OutputStreamWriter.java:254)
 	at org.postgresql.core.PGStream.flush(PGStream.java:590)
 	at org.postgresql.core.v3.QueryExecutorImpl.flushCopy(QueryExecutorImpl.java:1079)
 	... 17 more
 59489 [pool-1-thread-1] INFO org.apache.kafka.connect.storage.FileOffsetBackingStore  - Stopped FileOffsetBackingStore
 59490 [pool-1-thread-1] ERROR io.debezium.embedded.EmbeddedEngine  - Stopping connector after error in the application's handler method: java.lang.NullPointerException: null of tb.public.Message.Envelope
 org.apache.avro.file.DataFileWriter$AppendWriteException: java.lang.NullPointerException: null of tb.public.Message.Envelope
 	at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:308)
 	at io.materialize.tb.ChangeWriter.acceptInternal(ChangeWriter.java:97)
 	at io.materialize.tb.ChangeWriter.accept(ChangeWriter.java:71)
 	at io.materialize.tb.ChangeWriter.accept(ChangeWriter.java:42)
 	at io.debezium.embedded.ConvertingEngineBuilder.lambda$notifying$0(ConvertingEngineBuilder.java:70)
 	at io.debezium.embedded.EmbeddedEngine$1.handleBatch(EmbeddedEngine.java:471)
 	at io.debezium.embedded.EmbeddedEngine.run(EmbeddedEngine.java:810)
 	at io.debezium.embedded.ConvertingEngineBuilder$2.run(ConvertingEngineBuilder.java:170)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:834)
 Caused by: java.lang.NullPointerException: null of tb.public.Message.Envelope
 	at org.apache.avro.generic.GenericDatumWriter.npe(GenericDatumWriter.java:132)
 	at org.apache.avro.generic.GenericDatumWriter.writeWithoutConversion(GenericDatumWriter.java:126)
 	at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:73)
 	at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:60)
 	at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:302)
 	... 13 more
 Caused by: java.lang.NullPointerException
 	at org.apache.avro.generic.GenericData.getField(GenericData.java:666)
 	at org.apache.avro.generic.GenericData.getField(GenericData.java:681)
 	at org.apache.avro.generic.GenericDatumWriter.writeField(GenericDatumWriter.java:151)
 	at org.apache.avro.generic.GenericDatumWriter.writeRecord(GenericDatumWriter.java:143)
 	at org.apache.avro.generic.GenericDatumWriter.writeWithoutConversion(GenericDatumWriter.java:105)
 	... 16 more
```

got the same problem with mysql: 
```
tb_1            | 5312 [pool-1-thread-1] ERROR io.debezium.embedded.EmbeddedEngine  - Stopping connector after error in the application's handler method: java.lang.NullPointerException: null of tb.root.test1.Envelope
tb_1            | org.apache.avro.file.DataFileWriter$AppendWriteException: java.lang.NullPointerException: null of tb.root.test1.Envelope
tb_1            |       at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:308)
tb_1            |       at io.materialize.tb.ChangeWriter.acceptInternal(ChangeWriter.java:97)
tb_1            |       at io.materialize.tb.ChangeWriter.accept(ChangeWriter.java:71)
tb_1            |       at io.materialize.tb.ChangeWriter.accept(ChangeWriter.java:42)
tb_1            |       at io.debezium.embedded.ConvertingEngineBuilder.lambda$notifying$0(ConvertingEngineBuilder.java:70)
tb_1            |       at io.debezium.embedded.EmbeddedEngine$1.handleBatch(EmbeddedEngine.java:471)
tb_1            |       at io.debezium.embedded.EmbeddedEngine.run(EmbeddedEngine.java:810)
tb_1            |       at io.debezium.embedded.ConvertingEngineBuilder$2.run(ConvertingEngineBuilder.java:170)
tb_1            |       at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
tb_1            |       at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
tb_1            |       at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
tb_1            |       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
tb_1            |       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
tb_1            |       at java.base/java.lang.Thread.run(Thread.java:834)
tb_1            | Caused by: java.lang.NullPointerException: null of tb.root.test1.Envelope
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.npe(GenericDatumWriter.java:132)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.writeWithoutConversion(GenericDatumWriter.java:126)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:73)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.write(GenericDatumWriter.java:60)
tb_1            |       at org.apache.avro.file.DataFileWriter.append(DataFileWriter.java:302)
tb_1            |       ... 13 more
tb_1            | Caused by: java.lang.NullPointerException
tb_1            |       at org.apache.avro.generic.GenericData.getField(GenericData.java:666)
tb_1            |       at org.apache.avro.generic.GenericData.getField(GenericData.java:681)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.writeField(GenericDatumWriter.java:151)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.writeRecord(GenericDatumWriter.java:143)
tb_1            |       at org.apache.avro.generic.GenericDatumWriter.writeWithoutConversion(GenericDatumWriter.java:105)
tb_1            |       ... 16 more
```
As soon as I delete a row, tb panics.
Here is a minimal example I came up with (requires some manual steps tho). Still hope it helps:
https://github.com/docteurklein/mysql-tb-materialize

--- a/misc/tb/src/ChangeWriter.java
+++ b/misc/tb/src/ChangeWriter.java
@@ -94,7 +94,9 @@ public class ChangeWriter implements Consumer<RecordChangeEvent<SourceRecord>> {
         }
 
         Object datum = avroConverter.fromConnectData(record.valueSchema(), record.value());
-        dfw.append(datum);
+        if (datum != null) {
+            dfw.append(datum);
+        }
     }
 
     /**

